proposals/032-deprecate-unintended-apis.md

```diff
@@ -0,0 +1,221 @@
+
+* If a symbol can be moved into `slang-deprecated.h` without breaking other declarations in `slang.h`, it should be moved there
+
+The features we propose to deprecate are:
+
+* interface `ISlangCastable`
+
+* interface `ISlangClonable`
+
+* `struct SlangTerminatedChars`
+
+* type `SlangPathTypeIntegral`
+
+* `enum SlangPathType`
+
+* type `FileSystemContentsCallBack`
+
+* `enum OSPathKind`
+
+* `enum PathKind`
+
+* interface `ISlangFileSystemExt`
+
+* interface `ISlangMutableFileSystem`
+
+* type `SlangWriterChannelIntegral`
+
+* `enum SlangWriterChannel`
+
+* type `SlangWriterModeIntegral`
+
+* `enum SlangWriterMode`
+
+* interface `ISlangWriter`
+
+* interface `ISlangProfiler`
+
+* type `SlangDiagnosticCallback`
+
+* method `slang::IGlobalSession::addBuiltins`
+
+* method `slang::IGlobalSession::setDownstreamCompilerPrelude`
+
+* method `slang::IGlobalSession::getDownstreamCompilerPrelude`
+
+* method `slang::IGlobalSession::setLanguagePrelude`
+
+* method `slang::IGlobalSession::getLanguagePrelude`
+
+* method `slang::IGlobalSession::checkPassThroughSupport`
+
+* method `slang::IGlobalSession::compileCoreModule`
+
+* method `slang::IGlobalSession::loadCoreModule`
+
+* method `slang::IGlobalSession::saveCoreModule`
+
+* method `slang::IGlobalSession::saveCoreModule`
+
+* method `slang::IGlobalSession::getCompilerElapsedTime`
+
+* method `slang::IGlobalSession::compileBuiltinModule`
+
+* method `slang::IGlobalSession::loadBuiltinModule`
+
+* method `slang::IGlobalSession::saveBuiltinModule`
+
+* `enum slang::BuiltinModuleName`
+
+* `enum SlangArchiveType`
+
+* type `CompileCoreModuleFlags`
+
+* `struct CompileCoreModuleFlag`
+
+* method `slang::ISession::createCompileRequest`
+
+* method `slang::ISession::loadModuleFromIRBlob` should be renamed to `loadModuleFromBinaryBlob`
+
+* method `slang::IComponentType::getResultAsFileSystem`
+
+* method `slang::IComponentType::getEntryPointHostCallable`
+
+* function `slang_createGlobalSessionWithoutCoreModule`
+
+* function `slang_getEmbeddedCoreModule`
+
+* option `-emit-ir` should be deprecated
+
+* option `-report-downstream-time` should be marked as internal
+
+* option `-report-perf-benchmark` should be marked as internal
+
+* option `-report-checkpoint-intermediates` should be marked as internal
+
+* option `-source-embed-style` should be deprecated
+
+* option `-source-embed-name` should be deprecated
+
+* option `-source-embed-language` should be deprecated
+
+* option `-unscoped-enum` should be marked as undesirable (and subject to future deprecation/removal)
+
+* option `-preserve-params` should be marked as undesirable (and subject to future deprecation/removal)
+
+* option `-reflection-json` should be marked as undesirable (and subject to future deprecation/removal)
+
+* option `-msvc-style-bitfield-packing` should be marked as undesirable (and subject to future deprecation/removal)
+
+* option `-fspv-reflect` should be marked as undesirable (and subject to future deprecation/removal)
```

Comment:
Isn’t this required to support Khronos’s SPIRV reflection library? https://github.com/KhronosGroup/SPIRV-Reflect if I understand right, then IIRC Valve was a proponent that we support SPIRV-Reflect decorators.

 If this is true, then I’d probably object to removing this command line argument.